lib/qtrajectory.h

```diff
@@ -310,11 +341,13 @@ class QuantumTrajectorySimulator {
       }
     }
 
-    if (!ApplyDeferredOps(param, num_qubits, simulator, gates, state)) {
-      return false;
-    }
+    if (apply_last_deferred_ops || !stat.primary) {
```

Comment:
Yes, if `apply_last_deferred_ops` is false and the primary trajectory is realized then this block is skipped and the state doesn't evolve through the deferred ops. This leads to a speedup. Note that `apply_last_deferred_ops` should be set to false after the first primary trajectory is realized (the state is fully evolved for the first primary trajectory and the required results are collected by the client).

Reply:
So during the first primary trajectory, results are collected for all subsequent realizations of that trajectory? I don't see where this is happening. For example, to me it looks like `RunBatch` measures the fully-evolved primary trajectory once, then measures the partially-evolved trajectory (i.e. without deferred ops) for all subsequent primary realizations, which doesn't seem like it would produce valid results.

Reply:
Yes, right after the first primary trajectory, results should be collected for all subsequent realizations of that trajectory. The user should do that, see tests and apps/qsim_qtrajectory_cuda.cu. `RunBatch` “measures” after every trajectory, i.e. it calls `measure` after every trajectory. The user should collect the required results in `measure` after the first primary trajectory (when `stat.primary` is set to true for the first time). We are not interested in the fully-evolved state for all subsequent primary trajectories because the state is always the same.

Reply:
> The user should collect the required results in `measure` after the first primary trajectory {...}

Ahh, this is the part I was missing (and now see that it's mentioned in the docstring). Thanks for bringing me up to speed.